src/Umbraco.Core/Services/RelationService.cs

```diff
@@ -499,11 +499,28 @@ public bool HasRelations(IRelationType relationType)
     }
 
     /// <inheritdoc />
-    public bool IsRelated(int id)
+    public bool IsRelated(int id) => IsRelated(id, RelationDirectionFilter.Any);
+
+    /// <inheritdoc />
+    public bool IsRelated(int id, RelationDirectionFilter directionFilter)
     {
         using (ICoreScope scope = ScopeProvider.CreateCoreScope(autoComplete: true))
         {
-            IQuery<IRelation> query = Query<IRelation>().Where(x => x.ParentId == id || x.ChildId == id);
+            IQuery<IRelation> query = Query<IRelation>();
+
+            if (directionFilter == RelationDirectionFilter.Parent)
+            {
+                query = query.Where(x => x.ParentId == id);
+            }
+            else if (directionFilter == RelationDirectionFilter.Child)
+            {
+                query = query.Where(x => x.ChildId == id);
+            }
+            else
+            {
+                query = query.Where(x => x.ParentId == id || x.ChildId == id);
+            }
+
```

Comment:
Very minor thing, but I'm wondering if we should be more specific about the functionality here, just in case we end up adding to the `RelationDirectionFilter` in the future? Maybe even using a switch :smile: 

```suggestion
        query = directionFilter switch
        {
            RelationDirectionFilter.Parent => query.Where(x => x.ParentId == id),
            RelationDirectionFilter.Child => query.Where(x => x.ChildId == id),
            RelationDirectionFilter.Any => query.Where(x => x.ParentId == id || x.ChildId == id),
            _ => throw new ArgumentOutOfRangeException(nameof(directionFilter)),
        };
```